apps/nextjs/src/components/AppComponents/Chat/Chat/hooks/useAilaStreamingStatus.ts

```diff
@@ -16,46 +16,39 @@ export const useAilaStreamingStatus = ({
   isLoading: boolean;
   messages: Message[];
 }): AilaStreamingStatus => {
-  const possibleTypePatch = `"type":"patch"`;
-  const possibleTypePrompt = `"type":"prompt"`;
-
+  const prevStatusRef = useRef<AilaStreamingStatus | null>(null);
   const ailaStreamingStatus = useMemo<AilaStreamingStatus>(() => {
+    const moderationStart = `MODERATION_START`;
+    const chatStart = `CHAT_START`;
+    if (messages.length === 0) return "Idle";
     const lastMessage = messages[messages.length - 1];
-    const contentSplitByRs = lastMessage?.content.split("␞");
-    const lastSectionOfStreamingContent = contentSplitByRs
-      ? contentSplitByRs[contentSplitByRs?.length - 1]
-      : "";
+
     if (isLoading) {
       if (!lastMessage) return "Loading";
+      const { content } = lastMessage;
       if (lastMessage.role === "user") {
         return "RequestMade";
+      } else if (content.includes(moderationStart)) {
+        return "Moderating";
       } else if (
-        lastMessage.role === "assistant" &&
-        lastSectionOfStreamingContent?.includes(possibleTypePatch)
-      ) {
-        return "StreamingLessonPlan";
-      } else if (
-        lastMessage.content.includes(possibleTypePatch) &&
-        lastMessage.content.includes(possibleTypePrompt) &&
-        lastSectionOfStreamingContent !== ""
+        content.includes(`"type":"prompt"`) ||
+        content.includes('"type":"prompt"')
       ) {
         return "StreamingChatResponse";
-      } else if (
-        lastMessage.content.includes(possibleTypePatch) &&
-        lastMessage.content.includes(possibleTypePrompt) &&
-        lastSectionOfStreamingContent === ""
-      ) {
-        return "Moderating";
+      } else if (content.includes(chatStart)) {
+        return "StreamingLessonPlan";
       }
       return "Loading";
     }
     return "Idle";
-  }, [isLoading, messages, possibleTypePatch, possibleTypePrompt]);
+  }, [isLoading, messages]);
 
-  // @todo lets build a logger
-  if (process.env.NODE_ENV !== "development") {
-    console.log("ailaStreamingStatus", ailaStreamingStatus);
-  }
+  useEffect(() => {
+    if (prevStatusRef.current !== ailaStreamingStatus) {
+      console.log("ailaStreamingStatus changed:", ailaStreamingStatus);
+      prevStatusRef.current = ailaStreamingStatus;
+    }
+  }, [ailaStreamingStatus]);
```

Comment:
It looks like this could be replaced with a `useMemo`

Reply:
Hmm - are you supposed to have side-effects with useMemo?

Reply:
Oh, I see. `prevStatusRef` isn't used for anything, it's just to allow this effect to log. Sorry, I thought the goal was to derive one value from another

If you are ok with an initial log message, maybe you don't need a ref at all:

```ts
  useEffect(() => {
    console.log("ailaStreamingStatus set:", ailaStreamingStatus);
  }, [ailaStreamingStatus]);
```

Reply:
Good catch, thanks